### PR TITLE
mat64: add Reset method to SymDense

### PR DESCRIPTION
**Pull request by kortschak**
This is needed for an allocation improvement to stat.CovarianceMatrix and stat.CorrelationMatrix.

### CODE REVIEW

File: mat64/symmetric.go [hunk 1]
```diff
@@ -105,7 +105,20 @@ func (s *SymDense) SetRawSymmetric(b blas64.Symmetric) {
 	s.mat = b
 }
 
+// Reset zeros the dimensions of the matrix so that it can be reused as the
+// receiver of a dimensionally restricted operation.
+//
+// See the Reseter interface for more information.
+func (s *SymDense) Reset() {
+	// No change of Stride and N to 0
```
Comment (btracey) [inline mat64/symmetric.go]: Sorry, but I don't know what this comment is trying to tell me.

Comment (kortschak) [inline mat64/symmetric.go]: This is saying what all the Reset methods say, that you may not zero any of the listed fields without zeroing all of them.

Comment (btracey) [inline mat64/symmetric.go]: How about "Stride and N must be set to 0 together"?

**Comment by btracey** [comment]
LGTM with comment fix.

File: mat64/symmetric.go (hunk 1 above)
Comment (kortschak) [inline mat64/symmetric.go]: I will add commit that changes all three Reset comment to this PR. The wording needs thought.

**Comment by kortschak** [comment]
PTAL

**Comment by btracey** [comment]
Sorry, LGTM

**Pull request merged**